### PR TITLE
Add previously called `pbxproject_additional_elements` to `pbxproject_prefix`

### PR DESCRIPTION
**Pull request by brentleyjones**
After working towards what is now going to be called `target_dependencies`, I realized that `pbxproject_additional_elements` should be a part of `pbxproject_prefix`. A future change will merge `bazel_dependencies` into `pbxproject_prefix` as well, since they have very similar inputs.

I also renamed it to `pbxproj_prefix`.

### CODE REVIEW

**Comment by brentleyjones** [comment]
I'll add a `BuildSettingsTests` Monday morning.

File: tools/generators/pbxproject_prefix/src/Generator/BuildSettings.swift [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            "INDEXING_BUILT_PRODUCTS_DIR__YES":
+                "$(CONFIGURATION_BUILD_DIR)".pbxProjEscaped,
+            "INDEXING_DEPLOYMENT_LOCATION__":
+                "$(INDEXING_DEPLOYMENT_LOCATION__NO)".pbxProjEscaped,
+            "INDEXING_DEPLOYMENT_LOCATION__NO": "YES",
+            "INDEXING_DEPLOYMENT_LOCATION__YES": "NO",
+            "INDEXING_PROJECT_DIR__":
+                "$(INDEXING_PROJECT_DIR__NO)".pbxProjEscaped,
+            "INDEXING_PROJECT_DIR__NO": "$(PROJECT_DIR)".pbxProjEscaped,
+            "INDEXING_PROJECT_DIR__YES": "/Users/brentley/Developer/rules_xcodeproj/bazel-output-base/rules_xcodeproj.noindex/indexbuild_output_base/execroot/_main".pbxProjEscaped,
```
Comment (brentleyjones) [inline tools/generators/pbxproject_prefix/src/Generator/BuildSettings.swift:59]: FIXME

**Comment by brentleyjones** [comment]
@MobileNativeFoundation/rules_xcodeproj-maintainers This is good to go now.

File: tools/generators/pbxproject_prefix/src/Generator/BuildSettings.swift [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            "INDEXING_PROJECT_DIR__YES": indexingProjectDir.pbxProjEscaped,
+            "INDEX_FORCE_SCRIPT_EXECUTION": "YES",
+            "INSTALL_PATH":
+                "$(BAZEL_PACKAGE_BIN_DIR)/$(TARGET_NAME)/bin".pbxProjEscaped,
+            "INTERNAL_DIR": "$(PROJECT_FILE_PATH)/rules_xcodeproj",
+            "LD_DYLIB_INSTALL_NAME": "".pbxProjEscaped,
+            "LD_OBJC_ABI_VERSION": "".pbxProjEscaped,
+            "LD_RUNPATH_SEARCH_PATHS": "(\n\t\t\t\t)",
+            "ONLY_ACTIVE_ARCH": "YES",
+            // TODO: Set `RESOLVED_REPOSITORIES`
```
Comment (brentleyjones) [inline tools/generators/pbxproject_prefix/src/Generator/BuildSettings.swift:73]: We can't do this yet, we need input from the `files_and_groups` generator that will be made later.

File: tools/generators/pbxproject_prefix/src/Generator/BuildSettings.swift [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    ///   - workspace: The absolute path to the Bazel workspace.
+    static func buildSettings(
+        buildMode: BuildMode,
+        indexingProjectDir: String,
+        workspace: String
+    ) -> String {
+        var settings: [String: String] = [
+            "ALWAYS_SEARCH_USER_PATHS": "NO",
+            "BAZEL_CONFIG": "rules_xcodeproj",
+            "BAZEL_EXTERNAL": "$(BAZEL_OUTPUT_BASE)/external".pbxProjEscaped,
```
Comment (brentleyjones) [inline tools/generators/pbxproject_prefix/src/Generator/BuildSettings.swift:21]: I could escape all of these static strings manually, but I found it more readable this way.

File: tools/generators/pbxproject_prefix/README.md [hunk 4]
```diff
@@ -24,9 +27,14 @@ $ pbxproject_prefix \
...
     enGB \
-    --organization-name MobileNativeFoundation
+    --organization-name MobileNativeFoundation \
+    --xcode-configurations \
+    Debug \
+    Release \
+    --default-xcode-configuration Release
 ```
 
 ## Output
```
Comment (brentleyjones) [inline tools/generators/pbxproject_prefix/README.md:40]: I need to add to the example output

**Comment by brentleyjones** [comment]
I'm going to rename this to `pbxproj_prefix` now as well.

**Comment by brentleyjones** [comment]
Here is them merging together: https://github.com/MobileNativeFoundation/rules_xcodeproj/pull/2226

**Review by thiagohmcruz — APPROVED** [review]

File: tools/generators/README.md [hunk 5]
```diff
@@ -45,12 +42,16 @@ information.
 - `pbxproject_targets`:
   - The `PBXProject.targets` property
   - Closes the `PBXProject` element
+- `target_dependencies`:
+  - `PBXTargetDependency`s
+  - `PBXContainerItemProxy`s
 - [`bazel_dependencies`](bazel_dependencies/README.md):
   - All of the `BazelDependencies` related elements:
-    - `Generate Bazel Dependencies` script build phase
+    - `Bazel Build` script build phase
```
[the remark this answers is not in the document]
Comment (brentleyjones) [reply tools/generators/README.md:50]: Oops, I got it flipped. I don't want to make this change.

**Review by chiragramani — APPROVED** [review]

**Pull request merged**